crates/typst-pdf/src/gradient.rs

```diff
@@ -187,24 +204,24 @@ fn shading_function(ctx: &mut PdfContext, gradient: &Gradient) -> Ref {
                         .exponential_function(func1)
                         .range(gradient.space().range())
                         .c0(gradient.space().convert(first.0))
-                        .c1([1.0, s1 * (1.0 - t) + s2 * t, x1 * (1.0 - t) + x2 * t])
+                        .c1(components(if h1 < h2 { 1.0 } else { 0.0 }))
                         .domain([0.0, 1.0])
                         .n(1.0);
 
                     let func2 = ctx.alloc.bump();
                     ctx.pdf
                         .exponential_function(func2)
                         .range(gradient.space().range())
-                        .c0([1.0, s1 * (1.0 - t) + s2 * t, x1 * (1.0 - t) + x2 * t])
-                        .c1([0.0, s1 * (1.0 - t) + s2 * t, x1 * (1.0 - t) + x2 * t])
+                        .c0(components(1.0))
+                        .c1(components(0.0))
                         .domain([0.0, 1.0])
                         .n(1.0);
```

Comment:
I was wondering why this "zero-width" function exists when two partial gradients would theoretically suffice. In my test cases it was also working without it, but I assumed that there was a reason for this.

Reply:
I think I had run into an issue in some reader (can't remember which) and this fixed it, maybe I am misremembering, but there was a method to the madness